framework_tvb/tvb/adapters/simulator/model_forms.py

```diff
@@ -107,11 +108,26 @@ class StateVariableRangesForm(Form):
     def __init__(self):
         super(StateVariableRangesForm, self).__init__()
 
+class ModelForm(FormWithRanges):
 
-class Generic2dOscillatorModelForm(FormWithRanges):
+    def __init__(self, model, are_voi_disabled=False):
+        super(ModelForm, self).__init__()
 
-    def __init__(self):
-        super(Generic2dOscillatorModelForm, self).__init__()
+        self.variables_of_interest = MultiSelectField(model.get_class().variables_of_interest)
+        self.are_voi_disabled = are_voi_disabled
```

Comment:
It does not seem necessary to move the definition of `self.variables_of_interest` on `ModelForm`. You can keep here only the flag and the definition of `fill_from_trait` method.

Reply:
But doesn't it look better this way? We don't define variables_of_interest in every subclass + in the other fragments where they are more subclasses (integrators and monitors) the common param comes first (integration step_size and monitor sampling period) and only then come the specific params depending on the chosen type.

Reply:
But still, you need to send the model as a parameter from each subclass in order to define the variables_of_interest in a commonplace.
I would not change the order of the parameters if we can avoid it.

Reply:
Ok. Pycharm will give a warning, but it works like you say too.